### PR TITLE
NEXUS-4630, Changing DefaultNexusConfiguration to use ThreadContext.getSubject() so a subject is NOT created at boot.

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

**Comment by bdemers** [comment]
I have this ready to be pushed to 1.9.2.x

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java [hunk 1]
```diff
@@ -291,8 +293,12 @@ public class NexusHttpAuthenticationFilter
...
         {
             applicationEventMulticaster.notifyEventListeners( new NexusAuthenticationEvent( this,
-                new AuthenticationItem( username, RemoteIPFinder.findIP( (HttpServletRequest) request ), userAgent,
-                    success ) ) );
+                                                                                            new AuthenticationItem(
+                                                                                                username,
+                                                                                                RemoteIPFinder.findIP(
+                                                                                                    (HttpServletRequest) request ),
+                                                                                                userAgent,
+                                                                                                success ) ) );
```
Comment (nabcos) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java:301]:
```
    this
    is
    pretty
    ugly
    ;)
```

**Comment by nabcos** [comment]
looks sane, but I can't judge the functional change. +0.5?

**Comment by velo** [comment]
+1

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java (hunk 1 above)
Comment (bdemers) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java:301]: Agreed,  I think we need to resolve our formatting problems.

We need to be happy with some format, then format the whole code base.

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java [hunk 2]
```diff
@@ -48,6 +43,12 @@ import org.sonatype.nexus.rest.RemoteIPFinder;
 import org.sonatype.nexus.security.filter.NexusJSecurityFilter;
 import org.sonatype.plexus.appevents.ApplicationEventMulticaster;
 
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
```
Comment (cstamas) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/filter/authc/NexusHttpAuthenticationFilter.java:51]: Why was import order reordered?

**Pull request merged**